syntax = "proto3";

package chat;

service Chatter {
  rpc sendMessage(MessageRequest) returns (MessageRequest);
  rpc register(RegisterRequest) returns (StatusResponse);
  rpc findPair(FindPairRequest) returns (StatusResponse);
  rpc connectPair(ConnectPairRequest) returns (PairConnection);
  rpc getFromQueue(ReadQueueRequest) returns (stream MessageRequest);
}

message MessageRequest {
  string sender = 1;
  string receiver = 2;
  string message = 3;
}

message RegisterRequest {
  string user_name = 1;
}

message FindPairRequest {
  string target_name = 1;
}

message StatusResponse {
  uint32 responseCode = 1;
  string responseMessage = 2;
}

message ConnectPairRequest {
  string host_name = 1;
  string target_name = 2;
  uint32 connection_type = 3;
}

message PairConnection {
  string sender_name = 1;
  string receiver_name = 2;
  uint32 is_connection_created = 3;
}

message ReadQueueRequest {
  string read_from = 1;
}

